### PR TITLE
Implement irq_offload() on Cortex M0

### PR DESCRIPTION
**Pull request by tyrantlizard77**

### CODE REVIEW

File: arch/arm/core/irq_offload.c [hunk 1]
```diff
@@ -22,9 +22,17 @@ void _irq_do_offload(void)
 
 void irq_offload(irq_offload_routine_t routine, void *parameter)
 {
-	int key;
+#if defined(CONFIG_ARMV6_M) && defined (CONFIG_ASSERT)
```
Comment (lpereira) [inline arch/arm/core/irq_offload.c]: Minor nit, but space after the second `defined`.

**Review by nashif — DISMISSED** [review]
tested on multiple boards, LGTM

File: arch/arm/core/swap.S [hunk 2]
```diff
@@ -217,6 +220,17 @@ _thread_irq_disabled:
     bx lr
 
 #if defined(CONFIG_ARMV6_M)
+#ifdef CONFIG_IRQ_OFFLOAD
+SECTION_FUNC(TEXT, __svc)
+    push {lr}
```
Comment (MaureenHelm) [inline arch/arm/core/swap.S]: Shouldn't we parse the service call number first? Otherwise this will break if we hit a kernel_oops

Comment (andrewboie) [inline arch/arm/core/swap.S]: That's right, we use SVC for more than one thing now. Will fix.

Comment (andrewboie) [inline arch/arm/core/swap.S]: should be done now

**Review by cvinayak — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**